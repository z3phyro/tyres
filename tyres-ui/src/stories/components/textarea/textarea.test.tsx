import { expect, test, vi } from "vitest";
import { render } from "@solidjs/testing-library";
import Textarea from "./textarea";
import userEvent from "@testing-library/user-event";

test("Textarea renders", () => {
  const { getByRole } = render(() => <Textarea value="" onInput={() => { }} />);

  const textareaElement = getByRole("textbox");
  expect(textareaElement).toBeTruthy();
});

test("Textarea renders with label", () => {
  const { getByText } = render(() => <Textarea label="Test label" value="" onInput={() => { }} />);

  const labelElement = getByText("Test label");
  expect(labelElement).toBeTruthy();
});

test("Textarea renders with placeholder", () => {
  const { getByPlaceholderText } = render(() => <Textarea value="" onInput={() => { }} placeholder="Test placeholder" />);

  const placeholderElement = getByPlaceholderText("Test placeholder");
  expect(placeholderElement).toBeTruthy();
});

test("Textarea calls onInput", async () => {
  const onInput = vi.fn();
  const user = userEvent.setup();
  const { getByRole } = render(() => <Textarea value="" onInput={onInput} />);

  const textareaElement = getByRole("textbox");
  await user.type(textareaElement, "Test input");

  expect(onInput).toHaveBeenCalled();
});

test("Textarea renders with custom class", () => {
  const { getByRole } = render(() => <Textarea value="" onInput={() => { }} class="my-custom-class" />);

  const textareaElement = getByRole("group");
  expect(textareaElement).toHaveClass("my-custom-class");
});

test("Textarea renders with custom value", () => {
  const { getByRole } = render(() => <Textarea value="Test value" onInput={() => { }} />);

  const textareaElement = getByRole("textbox");
  expect(textareaElement).toHaveValue("Test value");
});

test("Textarea snapshot test", () => {
  const { asFragment } = render(() => <Textarea value="" onInput={() => { }} />);
  expect(asFragment()).toMatchSnapshot();
});
