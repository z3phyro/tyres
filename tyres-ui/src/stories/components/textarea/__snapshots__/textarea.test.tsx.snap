// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea snapshot test 1`] = `
"<div role="group" id="textfield-cl-12"><textarea id="textfield-cl-12-textarea" name="textfield-cl-12" class="w-full p-2 bg-white rounded mb-2 border border-1 outline-0 
        focus:shadow-md focus:shadow-blue-50 focus:border-blue-500 transition-all 
        duration-300 "></textarea></div>"
`;
